core/src/main/java/com/sap/ai/sdk/core/commons/ClientResponseHandler.java

```diff
@@ -18,28 +19,78 @@
 import org.apache.hc.core5.http.io.HttpClientResponseHandler;
 import org.apache.hc.core5.http.io.entity.EntityUtils;
 
+/**
+ * Parse incoming JSON responses and handles any errors
+ *
+ * @param <T> The type of the response
+ * @param <E> The type of the exception to throw
+ */
 @Slf4j
 @RequiredArgsConstructor
-class OrchestrationResponseHandler<T> implements HttpClientResponseHandler<T> {
+public class ClientResponseHandler<T, E extends ClientException>
+    implements HttpClientResponseHandler<T> {
```

Comment:
I'm marking this class as Beta, to discourage externals from using this class.

Maybe we can find a way to make this class not 100% dependent on Apache HTTP Client 5 API(?)